aqueduct_client/aqueduct_client.py

```diff
@@ -0,0 +1,243 @@
+            to load the key from a credentials file or from an
+            environment variable.
+
+        base_url : str, optional
+            The base URL for the Aqueduct API.  Defaults to the
+            FactSet Aqueduct endpoint.
+        """
+        if base_url:
+            self._base_url = base_url
+        else:
+            self._base_url = \
+                "https://factset.quantopian.com/api/experimental/pipelines"
+
+        if api_key is not None:
+            self._api_key = api_key
+        else:
+            self._api_key = load_api_key()
+
+    def get_all_pipelines(self):
+        """
+        Returns the metadata of all the pipelines you've run.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        List
+            A list of all the pipelines you have run.  Each pipeline is a dict
+            with id, start_date, end_date, created_at, status, and other
+            properties.
+        """
+        response = self._get('', None)
+        response.raise_for_status()
+        pipelines = response.json()['pipelines']
+        return pipelines
+
+    def get_pipeline(self, id):
+        """
+        Returns the metadata of a single pipeline.
+
+        Parameters
+        ----------
+        id : str
+            The id of the pipeline to load.
+
+        Returns
+        -------
+        dict
+            The metadata of the pipeline, containing id, start_date,
+            end_date, created_at, status, and other properties.
+        """
+        response = self._get('/{id}'.format(id=id), None)
+        response.raise_for_status()
+        pipeline = response.json()['pipeline']
+        return pipeline
+
+    def create_new_pipeline(self,
+                            code,
+                            start_date,
+                            end_date,
+                            name=None,
+                            params={},
+                            asset_identifier_format="symbol"):
+        """
+        Creates and queues a new pipeline execution.
+
+        Parameters
+        ----------
+        code : str
+            The pipeline code to run.
+        start_date : str
+            Execution start date, in YYYY-MM-DD format.
+        end_date : str
+            Execution end date, in YYYY-MM-DD format.
+        name : str (optional)
+            Human-readable name of the pipeline execution.
+        params : str (optional)
+            Input arguments for make_pipeline method defined in code.
+        asset_identifier_format : str (optional)
+            The type of identifier used to uniquely identify a security.
+            Valid options are "symbol", "sid", or "fsym_region_id".
+
+        Returns
+        ----------
+        {
+            "pipeline_id": str
+                The ID of the newly created pipeline.
+        }
+        """
+
+        try:
+            datetime.datetime.strptime(start_date, '%Y-%m-%d')
```

Comment:
Internally on Q, we generally use `pd.Timestamp(s)`, plus any additional required checks, as our preferred method for parsing datetimes. For consistency across environments, I would probably continue to use pandas here even if we want to normalize to a stdlib datetime eventually.

Reply:
ok, will switch to pandas for validating these date strings.